src/test/regress/sql/seclabel.sql

```diff
@@ -0,0 +1,105 @@
+--
+-- SECLABEL
+--
+-- Test suite for SECURITY LABEL ON ROLE statements
+--
+
+-- first we remove one of the worker nodes to be able to test
+-- citus_add_node later
+SELECT citus_remove_node('localhost', :worker_2_port);
+
+-- now we register a label provider
+CREATE FUNCTION citus_test_register_label_provider()
+  RETURNS void
+  LANGUAGE C
+  AS 'citus', $$citus_test_register_label_provider$$;
+SELECT citus_test_register_label_provider();
+
+CREATE ROLE user1;
+
+-- check an invalid label for our current dummy hook citus_test_object_relabel
+SECURITY LABEL ON ROLE user1 IS 'invalid_label';
+
+-- if we disable metadata_sync, the command will not be propagated
+SET citus.enable_metadata_sync TO off;
+SECURITY LABEL ON ROLE user1 IS 'citus_unclassified';
+SELECT objtype, objname, provider, label FROM pg_seclabels;
+
+\c - - - :worker_1_port
+SELECT objtype, objname, provider, label FROM pg_seclabels;
+
+\c - - - :master_port
+SELECT citus_test_register_label_provider();
+
+-- check that we only support propagating for roles
+SET citus.shard_replication_factor to 1;
+CREATE TABLE a (a int);
+SELECT create_distributed_table('a', 'a');
+SECURITY LABEL ON TABLE a IS 'citus_classified';
+DROP TABLE a;
+
+-- the registered label provider is per session only
+-- this means that we need to maintain the same connection to the worker node
+-- in order for the label provider to be visible there
+-- hence here we create the necessary session_level_connection_to_node functions
+SET citus.enable_metadata_sync TO off;
+CREATE OR REPLACE FUNCTION start_session_level_connection_to_node(text, integer)
+        RETURNS void
+        LANGUAGE C STRICT VOLATILE
+        AS 'citus', $$start_session_level_connection_to_node$$;
+CREATE OR REPLACE FUNCTION override_backend_data_gpid(bigint)
+        RETURNS void
+        LANGUAGE C STRICT IMMUTABLE
+        AS 'citus', $$override_backend_data_gpid$$;
+SELECT run_command_on_workers($$SET citus.enable_metadata_sync TO off;CREATE OR REPLACE FUNCTION override_backend_data_gpid(bigint)
+        RETURNS void
+        LANGUAGE C STRICT IMMUTABLE
+        AS 'citus'$$);
+CREATE OR REPLACE FUNCTION run_commands_on_session_level_connection_to_node(text)
+        RETURNS void
+        LANGUAGE C STRICT VOLATILE
+        AS 'citus', $$run_commands_on_session_level_connection_to_node$$;
+CREATE OR REPLACE FUNCTION stop_session_level_connection_to_node()
+        RETURNS void
+        LANGUAGE C STRICT VOLATILE
+        AS 'citus', $$stop_session_level_connection_to_node$$;
+RESET citus.enable_metadata_sync;
+
+-- now we establish a connection to the worker node
+SELECT start_session_level_connection_to_node('localhost', :worker_1_port);
```

Comment:
why does the citus_test_register_label_provider command need to be executed over the same connection as the follow up SECURITY LABEL one? Is it because if the function has not been called in a session we cannot use the label provider. 

One other approach could be using a GUC instead of a function, and if the GUC is true we trigger the provider registration from `_PG_init`.

Reply:
I'm not so sure if adding another GUC is better, even if it's for testing purposes.

Reply:
I think that's fair, we already have quite a few GUCs just for testing. I agree it's not great to add even more. But renaming `citus.running_under_isolation_test` to something like `citus.running_under_citus_test_suite` and then using it for both our isolation test locking stuff and this would be fine I think.

Reply:
One reason why I would prefer creating the provider in _PG_init is because it sounds like that's what all actual providers would do. So that way we test against something that's closer to reality.

Reply:
I agree with Jelte on this one, very good idea - especially the part that it will be actually like what an actual provider is like: loading the security label from PG_init. Since we already have one GUC specific for tests, I am extending that one.